zebra/interface.c

```diff
@@ -1729,64 +1859,62 @@ DEFPY (show_interface_vrf_all,
 
 /* Show specified interface to vty. */
 
-DEFUN (show_interface_name_vrf,
-       show_interface_name_vrf_cmd,
-       "show interface IFNAME vrf NAME",
-       SHOW_STR
-       "Interface status and configuration\n"
-       "Interface name\n"
-       VRF_CMD_HELP_STR)
+DEFPY(show_interface_name_vrf, show_interface_name_vrf_cmd,
+      "show [<ipv4|ipv6>$ip_version] interface [IFNAME$if_name] [vrf NAME$vrf_name]",
+      SHOW_STR IP_STR IPV6_STR
+      "Interface status and configuration\n"
+      "Interface name\n" VRF_CMD_HELP_STR)
 {
-	int idx_ifname = 2;
-	int idx_name = 4;
+	int ipa_type;
 	struct interface *ifp;
-	vrf_id_t vrf_id;
+	vrf_id_t vrf_id = VRF_DEFAULT;
 
 	interface_update_stats();
 
-	VRF_GET_ID(vrf_id, argv[idx_name]->arg, false);
+	if (vrf_name)
+		VRF_GET_ID(vrf_id, vrf_name, false);
+
+	ipa_type = parse_ipa_type(ip_version);
 
 	/* Specified interface print. */
-	ifp = if_lookup_by_name(argv[idx_ifname]->arg, vrf_id);
+	ifp = if_lookup_by_name(if_name, vrf_id);
 	if (ifp == NULL) {
-		vty_out(vty, "%% Can't find interface %s\n",
-			argv[idx_ifname]->arg);
+		vty_out(vty, "%% Can't find interface %s\n", if_name);
 		return CMD_WARNING;
 	}
-	if_dump_vty(vty, ifp);
+	if_dump_vty(vty, ifp, ipa_type);
 
 	return CMD_SUCCESS;
 }
 
 /* Show specified interface to vty. */
-DEFUN (show_interface_name_vrf_all,
-       show_interface_name_vrf_all_cmd,
-       "show interface IFNAME [vrf all]",
-       SHOW_STR
-       "Interface status and configuration\n"
-       "Interface name\n"
-       VRF_ALL_CMD_HELP_STR)
+DEFPY(show_interface_name_vrf_all, show_interface_name_vrf_all_cmd,
```

Comment:
I integrated this PR and tried testing. It looks like all of the below command are broken.

![image](https://user-images.githubusercontent.com/10162511/77208558-507bb500-6ab9-11ea-8fd8-9c372d7e7a7b.png)

Reply:
Thanks for the your testing and result.
I think it could be solved by changing DEFPY to DEFUN.